### DISCUSSION TITLE
Sound control?

### DISCUSSION THREAD

**Discussion by KeyboardZombie** [greghesp/ha-bambulab#815] [Ideas]
Hi, I can't find anyway to turn off and on sounds from my a1 mini in this. Would be useful as I don't want prints that finish in the middle of the night making noise, but like the noise during the day. thanks.

**Comment by AdrianGarside** [comment]
Can you please try this version?
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.40-a1sound2

I coded it up ages ago but I have no way to test and the person who asked for it vanished.

**Reply by KeyboardZombie** [reply]
Sure, I've got it updated to this version but out rn so will update you later.

**Reply by AdrianGarside** [reply]
I lost the switch name during the merge conflict resolution. That's fixed in:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.40-a1sound6

The version you have should still function, but the switch name will incorrectly be 'prompt_sound'

**Comment by AdrianGarside** [comment]


**Reply by KeyboardZombie** [reply]
The sound option is listed as bambu 3d printer but says its unavailible 
![Screenshot_20250101_171621_Home Assistant](https://github.com/user-attachments/assets/7e267e00-4431-4cc3-9e4a-5dd645e0f9a3)
switch.bambu_3d_printer_none ⸱ Bambu Lab ⸱

**Reply by AdrianGarside** [reply]
Which version do you have installed?

**Reply by KeyboardZombie** [reply]
I installed 2, didnt get a chance to play around with anything, then installed 6

**Reply by AdrianGarside** [reply]
The _none entry is probably due to the bug in 2 followed by upgrading to a build with the fix. Is there another switch that’s properly named?

**Reply by KeyboardZombie** [reply]
i deleted and re added my printer, can't see anything about sound now. also resume printing and manual refresh mode seem to be unavailable on a1 mini

**Reply by AdrianGarside** [reply]
Can you grab me a diagnostics log please?

**Reply by AdrianGarside** [reply]
You should be seeing a new switch in the configuration section:
![image](https://github.com/user-attachments/assets/5891c32a-f495-4fbf-bce6-8999353e4ad9)

**Comment by KeyboardZombie** [comment]
1 January 2025
A1mini_0309BA461703204 End time changed to 2025-01-01 22:42:00
22:31:16 - 2 minutes ago
A1mini_0309BA461703204 End time changed to 2025-01-01 22:41:00
22:29:39 - 4 minutes ago
A1mini_0309BA461703204 End time changed to 2025-01-01 22:42:00
22:29:10 - 5 minutes ago
A1mini_0309BA461703204 Speed profile changed to Ludicrous
22:29:03 - 5 minutes ago
A1mini_0309BA461703204 Printing speed changed to Ludicrous triggered by action Select: Select
22:29:03 - 5 minutes ago - Tom
A1mini_0309BA461703204 End time changed to 2025-01-01 22:50:00
22:25:55 - 8 minutes ago
A1mini_0309BA461703204 End time changed to 2025-01-01 22:51:00
22:24:02 - 10 minutes ago
A1mini_0309BA461703204 End time changed to 2025-01-01 22:50:00
22:22:58 - 11 minutes ago
A1mini_0309BA461703204 Aux fan turned on
22:21:20 - 12 minutes ago
A1mini_0309BA461703204 Cooling fan turned on
22:21:20 - 12 minutes ago
A1mini_0309BA461703204 Active tray changed to Bambu PLA Matte
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Print bed type changed to Textured plate
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Active tray index changed to 255
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 End time changed to 2025-01-01 22:51:00
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Start time changed to 2025-01-01 22:13:00
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Current stage changed to Printing
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Printing speed changed to Standard
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Chamber light turned on
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Cover Image changed to 1 January 2025 at 22:21
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Stop printing became unknown
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Pause printing became unknown
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Nozzle type changed to Stainless steel
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Print bed type became unknown
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Printer name changed to Bambu3D
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Print type changed to Cloud
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Task name changed to FISH EYE LENS.stl
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Gcode filename changed to FISH EYE LENS.stl.3mf
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Active tray index changed to 1
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Print status changed to Running
22:21:19 - 12 minutes ago
A1mini_0309BA461703204 Current stage changed to Idle
22:21:19 - 12 minutes ago
![diag](https://github.com/user-attachments/assets/38663ae1-2118-49f6-b543-b263b1e03f3f)
![{ABA17E39-1D8A-4486-A959-F1C2242942FE}](https://github.com/user-attachments/assets/0c9defa3-5eda-4a24-b529-0f3d11634e8b)

**Reply by AdrianGarside** [reply]
Please can you get a diagnostic log: 
![image](https://github.com/user-attachments/assets/17e4f27a-9a57-4339-8c4d-f0d2802788f9)

**Reply by KeyboardZombie** [reply]
got you, sorry. 
[config_entry-bambu_lab-01JGHZAX1VGKHCTSWAPA5SYNYC.json](https://github.com/user-attachments/files/18286843/config_entry-bambu_lab-01JGHZAX1VGKHCTSWAPA5SYNYC.json)

**Reply by AdrianGarside** [reply]
Ok I see what's going on. This should fix a bunch of your missing stuff:
https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.40-a1sound7

**Reply by KeyboardZombie** [reply]
> Ok I see what's going on. This should fix a bunch of your missing stuff: https://github.com/greghesp/ha-bambulab/releases/tag/v2.0.40-a1sound7

updated, still no sound option, it did change from 50 to 51 entities, but that may of been me playing with a camera setting, should i delete and re-add the printed?

**Reply by AdrianGarside** [reply]
Can you provide an updated diagnostic log?

**Reply by AdrianGarside** [reply]
And then try deleting and re-adding the printer.

**Reply by KeyboardZombie** [reply]
> Can you provide an updated diagnostic log?

[config_entry-bambu_lab-01JGHZAX1VGKHCTSWAPA5SYNYC.json](https://github.com/user-attachments/files/18288865/config_entry-bambu_lab-01JGHZAX1VGKHCTSWAPA5SYNYC.json)

**Reply by AdrianGarside** [reply]
This is weird. My changes should have fixed you and I can't see how they wouldn't. Are you also still missing the manual refresh switch?
